### PR TITLE
Added port to config

### PR DESCRIPTION
**Pull request by SeppPenner**
Summary of the changes (in less than 80 chars)

 * Adds a configuration option to the BaGet server to specify the port it runs on, check `appsettings.json` --> `Port`.

Fixes https://github.com/loic-sharma/BaGet/issues/485.

### CODE REVIEW

File: src/BaGet/Program.cs [hunk 1]
```diff
@@ -38,29 +40,27 @@ public static async Task Main(string[] args)
             {
                 var host = CreateWebHostBuilder(args).Build();
 
+                // Todo: Don't know how to get the options without rebuilding the host!
```
Comment (SeppPenner) [inline src/BaGet/Program.cs:43]: I don't know how to get the configuration here in another way...

**Comment by SeppPenner** [comment]
The only thing to check here is that I don't know how to get the configuration in program without building the host twice: https://github.com/loic-sharma/BaGet/blob/167fc22f0d6ecc3a5db4be9d180d4cfdefc2144c/src/BaGet/Program.cs#L43

File: src/BaGet/Program.cs (hunk 1 above)
Comment (loic-sharma) [reply src/BaGet/Program.cs:43]: I would follow this guide to capture configurations at startup: https://docs.microsoft.com/en-us/aspnet/core/fundamentals/host/web-host?view=aspnetcore-3.1#override-configuration

I would grab the `IConfigurationRoot`, look for a `Urls` config. If it exists, call [`UseUrls`](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/host/generic-host?view=aspnetcore-3.1#urls).

File: src/BaGet/Program.cs [hunk 2]
```diff
@@ -38,29 +40,27 @@ public static async Task Main(string[] args)
...
+                var baGetOptions = host.Services.GetRequiredService<IOptions<BaGetOptions>>();
+                host = CreateWebHostBuilder(args).UseUrls(baGetOptions.Value.Port).Build();
```
Comment (loic-sharma) [inline src/BaGet/Program.cs:45]: WebHost is the legacy host, the generic host is the preferred host. You should be able to use this instead: https://docs.microsoft.com/en-us/aspnet/core/fundamentals/host/generic-host?view=aspnetcore-3.1#urls

```cs
Host.CreateDefaultBuilder(args)
    .ConfigureWebHostDefaults(webBuilder =>
    {
        // configure kestrel...

        if (has urls config)
        {
            web.UseUrls(urls config);
        }

        web.UseStartup<Startup>();
    });
```

File: src/BaGet/appsettings.json [hunk 3]
```diff
@@ -1,5 +1,6 @@
 {
   "ApiKey": "",
+  "Port": "http://*:5000",
```
Comment (loic-sharma) [inline src/BaGet/appsettings.json:3]: Let's stay consistent with ASP.NET Core's terminology. I think `Urls` is a better config name since it better captures the functionality exposed here.

File: src/BaGet.Core/Configuration/BaGetOptions.cs [hunk 4]
```diff
@@ -38,6 +38,11 @@ public class BaGetOptions
         /// </summary>
         public bool IsReadOnlyMode { get; set; } = false;
 
+        /// <summary>
+        /// The port the baGet server will run on.
```
Comment (loic-sharma) [inline src/BaGet.Core/Configuration/BaGetOptions.cs:42]: Should we link to the documentation to explain the supported URL formats here? Specifically, this documentation: https://docs.microsoft.com/en-us/aspnet/core/fundamentals/host/web-host?view=aspnetcore-3.1#server-urls

File: src/BaGet/Program.cs [hunk 5]
```diff
@@ -4,63 +4,79 @@
...
 using Microsoft.AspNetCore.Hosting;
 using Microsoft.Extensions.DependencyInjection;
+using System.IO;
+using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.Hosting;
 
 namespace BaGet
 {
-    public class Program
+    namespace BaGet
```
Comment (loic-sharma) [inline src/BaGet/Program.cs:13]: Do we need to redeclare the namespace?

File: src/BaGet/Program.cs [hunk 6]
```diff
@@ -4,63 +4,79 @@
...
-                Name = "baget",
-                Description = "A light-weight NuGet service",
-            };
+                var app = new CommandLineApplication
+                {
+                    Name = "baget",
+                    Description = "A light-weight NuGet service",
+                };
 
-            app.HelpOption(inherited: true);
```
Comment (loic-sharma) [inline src/BaGet/Program.cs:21]: Please keep the parameter name. The .NET Foundation style is to provide the parameter name for any boolean/integer literal value.

File: src/BaGet/Program.cs [hunk 7]
```diff
@@ -4,63 +4,79 @@
...
+                            // Remove the upload limit from Kestrel. If needed, an upload limit can
+                            // be enforced by a reverse proxy server, like IIS.
+                            options.Limits.MaxRequestBodySize = null;
+                        });
 
-                    web.UseStartup<Startup>();
-                });
+                        var config = new ConfigurationBuilder()
+                            .SetBasePath(Directory.GetCurrentDirectory())
+                            .AddJsonFile("appsettings.json", true)
```
Comment (loic-sharma) [inline src/BaGet/Program.cs:63]: Please add a named parameter for the boolean literal:

```suggestion
                            .AddJsonFile("appsettings.json", optional: true)
```

File: src/BaGet.Hosting/Extensions/IHostExtensions.cs [hunk 8]
```diff
@@ -2,8 +2,9 @@
 using System.Threading.Tasks;
 using BaGet.Core;
 using Microsoft.Extensions.DependencyInjection;
-using Microsoft.Extensions.Hosting;
+using Microsoft.AspNetCore.Hosting;
 using Microsoft.Extensions.Options;
+using Microsoft.Extensions.Hosting;
```
Comment (loic-sharma) [inline src/BaGet.Hosting/Extensions/IHostExtensions.cs:7]: Can we revert these changes?

File: src/BaGet/Startup.cs [hunk 9]
```diff
@@ -5,8 +5,8 @@
 using Microsoft.AspNetCore.Hosting;
 using Microsoft.AspNetCore.SpaServices.ReactDevelopmentServer;
 using Microsoft.Extensions.Configuration;
-using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.DependencyInjection;
```
Comment (loic-sharma) [inline src/BaGet/Startup.cs:9]: Can we revert this change? This is no longer sorted alphabetically

File: src/BaGet/Program.cs [hunk 10]
```diff
@@ -4,63 +4,79 @@
 using McMaster.Extensions.CommandLineUtils;
 using Microsoft.AspNetCore.Hosting;
 using Microsoft.Extensions.DependencyInjection;
+using System.IO;
```
Comment (loic-sharma) [inline src/BaGet/Program.cs:7]: Super nit: please put `System.*` first, then alphabetical.

File: src/BaGet.UI/package-lock.json [hunk 11]
```diff
@@ -7162,7 +7162,8 @@
             },
             "ansi-regex": {
               "version": "2.1.1",
-              "bundled": true
+              "bundled": true,
+              "optional": true
```
Comment (loic-sharma) [inline src/BaGet.UI/package-lock.json:7166]: Let's revert these changes as we're not doing any UI changes as part of this PR.

**Comment by loic-sharma** [comment]
This looks great! I left some nitpicks, but after that I think it's ready to merge :)

File: src/BaGet/Program.cs (hunk 5 above)
Comment (SeppPenner) [reply src/BaGet/Program.cs]: No, I guess, this was some strange kind of copy paste error...

File: src/BaGet/Program.cs (hunk 6 above)
Comment (SeppPenner) [reply src/BaGet/Program.cs:21]: I guess, we really need a code style checker now 😄  That wasn't really intended, just Resharper removing everything that is not necessarily needed. I will use VS2019 preview (which doesn't have any extensions enabled to avoid these kind of errors in the future 👍

File: src/BaGet/Program.cs (hunk 7 above)
Comment (SeppPenner) [reply src/BaGet/Program.cs]: Is updated :)

File: src/BaGet/Startup.cs (hunk 9 above)
Comment (SeppPenner) [reply src/BaGet/Startup.cs:9]: Yep.

File: src/BaGet/Program.cs (hunk 10 above)
Comment (SeppPenner) [reply src/BaGet/Program.cs:7]: Yep :)

File: src/BaGet.Hosting/Extensions/IHostExtensions.cs (hunk 8 above)
Comment (SeppPenner) [reply src/BaGet.Hosting/Extensions/IHostExtensions.cs:7]: Sure.

File: src/BaGet.UI/package-lock.json (hunk 11 above)
Comment (SeppPenner) [reply src/BaGet.UI/package-lock.json:7166]: I have no idea how this was changed, honestly. Maybe I was working on the wrong branch or something 😅

**Comment by SeppPenner** [comment]
Now it looks good (and pretty clean) I would say. Sorry for the big amount of commits here but it's a lot easier to change one thing by one commit to not forget anything...

**Comment by SeppPenner** [comment]
This probably addressed https://github.com/loic-sharma/BaGet/pull/525 as well (Can be closed, too after merge).

**Comment by loic-sharma** [comment]
> Now it looks good (and pretty clean) I would say.

Yup this looks great. Thanks for the awesome contribution! 

> Sorry for the big amount of commits here but it's a lot easier to change one thing by one commit to not forget anything...

No worries, I squash commits when I merge changes, so feel free to do as many commits as you'd like.

**Comment by SeppPenner** [comment]
> No worries, I squash commits when I merge changes, so feel free to do as many commits as you'd like.

Haha ok :)
